tools/node.d.ts

```diff
@@ -204,4 +210,915 @@ export class Dictionary {
+    negate: Function;
+    drop_side_effect_free: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_Defun extends AST_Lambda {
+    reduce_vars: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_Class extends AST_Scope {
+    _walk: Function;
+    transform: Function;
+    is_block_scope: Function;
+    _codegen: Function;
+    add_source_map: Function;
+    _eval: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+    is_constant_expression: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_DefClass extends AST_Class {
+    reduce_vars: Function;
+    has_side_effects: Function;
+}
+
+declare class AST_ClassExpression extends AST_Class {
+    needs_parens: Function;
+    reduce_vars: Function;
+    drop_side_effect_free: Function;
+}
+
+declare class AST_Switch extends AST_Block {
+    _walk: Function;
+    transform: Function;
+    _codegen: Function;
+    add_source_map: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_SwitchBranch extends AST_Block {
+    is_block_scope: Function;
+    _do_print_body: Function;
+    add_source_map: Function;
+    aborts: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_Default extends AST_SwitchBranch {
+    _codegen: Function;
+    reduce_vars: Function;
+}
+
+declare class AST_Case extends AST_SwitchBranch {
+    _walk: Function;
+    transform: Function;
+    _codegen: Function;
+    reduce_vars: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+}
+
+declare class AST_Try extends AST_Block {
+    _walk: Function;
+    transform: Function;
+    _codegen: Function;
+    add_source_map: Function;
+    reduce_vars: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_Catch extends AST_Block {
+    _walk: Function;
+    transform: Function;
+    _codegen: Function;
+    add_source_map: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_Finally extends AST_Block {
+    _codegen: Function;
+    add_source_map: Function;
+}
+
+declare class AST_EmptyStatement extends AST_Statement {
+    _codegen: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_StatementWithBody extends AST_Statement {
+    _do_print_body: Function;
+    add_source_map: Function;
+}
+
+declare class AST_LabeledStatement extends AST_StatementWithBody {
+    _walk: Function;
+    clone: Function;
+    transform: Function;
+    _codegen: Function;
+    add_source_map: Function;
+    reduce_vars: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_IterationStatement extends AST_StatementWithBody {
+    clone: Function;
```

Comment:
only declare `clone(boolean?);` on the `AST_Node` class. Not any derived classes.

Reply:
`clone` returns `AST_Node`.